parser/parser.go

```diff
@@ -32,6 +33,10 @@ var (
 	// Version2 is an alias for VersionGROQ1.
 	// Deprecated: Use VersionGROQ1.
 	Version2 Version = 2
+
+	// VersionX is a copy of Version2 with experimental support for GROQ fragments.
+	// Experimental: This is only an experimental version and may change at any time.
+	VersionX Version = 0
```

Comment:
nit:
```suggestion
	// VersionX is an experimental version of the GROQ parser. There's no guarantees that it will be stable and might change at any time.
	VersionX Version = 0
```